crypto/cms/cms_sd.c

```diff
@@ -765,12 +777,17 @@ static int cms_SignerInfo_content_sign(CMS_ContentInfo *cms,
             ERR_raise(ERR_LIB_CMS, ERR_R_MALLOC_FAILURE);
             goto err;
         }
-        if (!EVP_SignFinal_ex(mctx, sig, &siglen, si->pkey,
-                              ossl_cms_ctx_get0_libctx(ctx),
-                              ossl_cms_ctx_get0_propq(ctx))) {
-            ERR_raise(ERR_LIB_CMS, CMS_R_SIGNFINAL_ERROR);
-            OPENSSL_free(sig);
-            goto err;
+        if (precomp_md != NULL) {
```

Comment:
And here as well.

Reply:
Here, sig is allocated and a pointer to it is retained in si->signature afterwards. I think we'll have a problem about who is responsible for freeing precomp_md afterwards.

In the other cases above, md is stack allocated so there is no freeing involved. The caller has to free precomp_md.

I suggest keeping the memcpy in this case, but removing it in the other cases above. WDYT?

Reply:
Yes, I did not notice the pointer is kept here. So yeah, here memcpy has to be kept.